dvc/repo/plots/__init__.py

```diff
@@ -519,7 +524,7 @@ def unpack_if_dir(fs, path, props: dict[str, str], onerror: Optional[Callable] =
 
     if "data" in unpacked:
         for subpath in unpacked["data"]:
-            result["data"].update({subpath: props})
+            result["data"].update({subpath: {} | props})  # avoid shared reference
```

Comment:
You can use `.copy()` to make your intentions clearer for doing a shallow copy. No need of a comment.
```suggestion
            result["data"].update({subpath: props.copy()})
```

Reply:
Thanks for the suggestion :) 

Yeah we need to copy this `props` reference. Otherwise I ended up with cases like this:

architecture:
```
metrics/
    subdir/
        wonderful_plot.tsv
    super_plot.tsv
```


dvc.yaml
```yaml
plots:
    metrics:
        x: step
    metrics/subdir:
        x: timestamp
```

The first call to this function set all the plot configurations for the `metrics` folder (wonderful and super plots). Which is what we want. The second call, you expect that it only changes the configs for the content of `metrics/subdir` (in that case `super_plot.tsv`. It does so :tada: , but as he mutate  the same dictionary reference (`props`), the `wonderful_plot` configuration ends up changes to `x: timestamp` too. :crying_cat_face: